### PR TITLE
Make `fleet-dark` accurate to the official theme

### PR DESCRIPTION
**Pull request by kirawi**
# Objective

The current palette for the theme didn't use the official palette, and some scopes used the wrong colors. The contrast between the cursorline and comments also caused eye strain.

## Solution

I copied the palette from `dark.json` in the themes folder, and reworked the theme to reflect the official one. I'm a bit displeased with the color for `type.enum.variant`, which the official theme doesn't highlight, as well as the contrast between `variables.builtin` and the background (for which I preferred the old color).

Fleet vs. `new fleet-dark` (top) vs. `old fleet-dark` (bottom)

Capture
Capture1

---

## Changelog

updated `fleet-dark`'s palette to be identical to the one in the official theme

### CODE REVIEW

**Comment by kirawi** [comment]
\cc @krfl

**Comment by archseer** [comment]
Comparison screenshots?

**Comment by kirawi** [comment]
Yep, I've updated the description with them.

**Comment by krfl** [comment]
@kirawi Thank you for taking the time to do all this work. 

I have three comments:
1. The light grey background on the horizontal divider used by the pickers looks a little out of place compared to the rest of the theme.
Horizontal divider

2. The curly underlines were just changed in #5419. Official Fleet uses thick straight lines to underline errors. My understanding was that all themes that uses underlines should use curly underlines and I see that these were removed by your update. If we are going to show underlines, then it's going to be a compromise. It would be great to get a final confirmation on this. @pascalkuthe? 
(old `fleet_dark` on the left, new `fleet_dark` on the right)
Screenshot 2023-01-21 at 09 08 22

3. Please do add yourself to the list of contributors :)

**Comment by kirawi** [comment]
Thanks for the review! I didn't notice the background nor the missing underline for diagnostics. I've also diverged from the official theme by making `ui.cursor.match` yellow because the blue was too low-contrast.

File: runtime/themes/fleet_dark.toml
Comment (kirawi) [inline runtime/themes/fleet_dark.toml:70]: I commented out these since I didn't know what would be good colors.

Comment (krfl) [reply runtime/themes/fleet_dark.toml:70]:
```suggestion
"ui.statusline" = { fg = "Gray 120", bg = "Gray 20" }
"ui.statusline.insert" = { bg = "Blue 80" }
"ui.statusline.select" = { bg = "Yellow 40" }
```

I think it's important to separate `fleet` features from `helix` features and just make an educated guess at what people will prefer. Fleet relies heavily on blue, so I think it will be a safe and familiar color for insert mode. 

Yellow 40 or Green 40 might work too. The yellow is darker and warmer and reminds me of vim, but the green might be more in the style of fleet. What do you think?

**Review by krfl — COMMENTED** [review]
I would very much to know whether all themes should use curly lines or not, because that's the understanding I got from #5419

Also, I think that `Gray 80` for hints might be too subtle? It's not that it's hard to see or read, but I feel like it should pop a lot more. 

Here i have the old theme on the left and this update on the right. Also a comparison screenshot of fleet below. 
diagnostic

Screenshot 2023-01-23 at 00 36 02

**Comment by kirawi** [comment]
IIRC curly underlines were only applied to all themes that didn't explicitly specify the line type.

**Comment by pascalkuthe** [comment]
I think the curly part is less important as long as the underlines are colored so its possible to tell whether we are looking at a warning or an error.

**Comment by krfl** [comment]
> I checked the source again, and it looks like `Gray 80` is the color Fleet uses. Do you want to diverge from the official theme and use a higher contrast color?
> [dark.json.txt](https://github.com/helix-editor/helix/files/10476907/dark.json.txt)

I agree that `Gray 80` is correct in terms of matching what Fleet uses. I also believe that the statusbar is more important in Helix than in Fleet. The answer depends on whether you view the theme as an adaptation for Helix or as a copy of Fleet Dark.

Since I'm so torn, I'll let you decide. I will be okay either way. Promise :)

**Comment by kirawi** [comment]
Personally, I feel that it should be accurate to the official theme so it's not jarring for users. Users can customize the theme if they prefer :P

**Comment by kirawi** [comment]
Secondary selection highlighting is actually pretty bad, so I'll have to fix that before this could get merged.

**Comment by kirawi** [comment]
Fixed. I misread 50 as 80.

File: runtime/themes/fleet_dark.toml
Comment (the-mikedavis) [inline runtime/themes/fleet_dark.toml:57]: looks like this `purple` isn't defined anymore

**Review by the-mikedavis — APPROVED** [review]

**Pull request merged**
